maint/mpi_standard_api.txt

```diff
@@ -150,15 +150,15 @@ MPI_Alltoallw_init:
     info: INFO, [info argument]
     request: REQUEST, direction=out, [communication request]
 MPI_Attr_delete:
-    comm: COMMUNICATOR, direction=lis:inout, param:in, [communicator to which attribute is attached]
+    comm: COMMUNICATOR, direction=in, [communicator to which attribute is attached]
```

Comment:
has the script used to scrape the standard data been committed to the repo?

Reply:
No. 

The mpi standards -- at least the API functions -- are not expected to change often. I think it is simpler to directly maintain `mpi_standard_api.txt` directly (even manually), rather than having to maintain one more script and add a dependency on the private mpi-standards repository.

Reply:
I don't think this is a good idea. There is a hidden step to this that only you currently understand.

Reply:
If I say I manually transcribed the api interface into this config file, would that make any difference?

The script I use is in Perl and it is for one-time purpose that I don't intend to maintain (other than hack). This config file will always need be directly maintained anyway since we want to review explicit changes from this point on. If we get initial transcription correct, I expect it is not a big task in the future to manually update this file.

It is one of the intentions to decouple this config file from the python-framework used in mpi-standards. That python layer are subject to change with no apparent reasons from outside. For example, they may change the way how the categorize parameters, variable names, particular attributes syntaxes, etc.. I don't think we really want to follow those changes. Any changes to `mpi_standard_api.txt` in the future should be treated as an update from standards from user point of view.

Reply:
Besides, if we want a *correct* scraper, we'll need the scraper to be able to parse full Python syntax. That is way too complex and unnecessary. But without the *correctness*, the script is considered hacky and thus I am unwilling to maintain.

Reply:
> It is one of the intentions to decouple this config file from the python-framework used in mpi-standards.

I disagree with this intent. The source data in the MPI standard is json. Python includes a json parser in the standard library. I don't think this is a big ask.

Reply:
It is not json. It is full python code:
```
\begin{mpi-binding}
    function_name("MPI_Send")

    parameter(
        "buf", "BUFFER", desc="initial address of send buffer", constant=True
    )
    parameter(
        "count",
        "XFER_NUM_ELEM_NNI_SMALL",
        desc="number of elements in send buffer",
    )
    parameter(
        "datatype", "DATATYPE", desc="datatype of each send buffer element"
    )
    parameter("dest", "RANK", desc="rank of destination")
    parameter("tag", "TAG", desc="message tag")
    parameter("comm", "COMMUNICATOR")
\end{mpi-binding}
```

To do it "correctly", not only we need parse python syntax, we also need import all their python stack. Probably that the only way is to directly import their code, then we are having a hard dependency on it.

Reply:
If you build the document, there is an `apis.json` file generated.

Reply:
Also their python code is focused on generating 'LaTeX', so importing it still doesn't reduce the complexity much on our part.

Reply:
I'm not talking about importing their Python code. Please don't misunderstand me.

Reply:
I didn't realize the generated `apis.json`. We can directly use the json file -- i.e. directly add `apis.json` to our repository. Is that more preferable. The apiis.json doesn't have language bindings (parameter types), so that still need be separately scraped.

Reply:
It might be worth engaging the standard maintainers to see if parameter types might be something they can add to the json, or if some other solution exists.

I don't want to block others, so I think we can move forward with this. The API source information is a big piece of technical debt that I think we should address in the long-term.